src/SBInterpolatedImage.cpp

```diff
@@ -900,6 +782,198 @@ namespace galsim {
+                                                    kInterp, gsparams)
+        ) {}
+
+    SBInterpolatedKImage::SBInterpolatedKImage(const SBInterpolatedKImage& rhs)
+        : SBInterpolated(rhs) {}
+
+    SBInterpolatedKImage::~SBInterpolatedKImage() {}
+
+    ConstImageView<double> SBInterpolatedKImage::getRealKImage() const
+    {
+        assert(dynamic_cast<const SBInterpolatedKImageImpl*>(_pimpl.get()));
+        return static_cast<const SBInterpolatedKImageImpl&>(*_pimpl).getRealKImage();
+    }
+
+    ConstImageView<double> SBInterpolatedKImage::getImagKImage() const
+    {
+        assert(dynamic_cast<const SBInterpolatedKImageImpl*>(_pimpl.get()));
+        return static_cast<const SBInterpolatedKImageImpl&>(*_pimpl).getImagKImage();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    // SBInterpolatedKImageImpl methods
+
+    template <typename T>
+    SBInterpolatedKImage::SBInterpolatedKImageImpl::SBInterpolatedKImageImpl(
+        const BaseImage<T>& realKImage, const BaseImage<T>& imagKImage,
+        double dk, double stepk, boost::shared_ptr<Interpolant2d> kInterp,
+        const GSParamsPtr& gsparams) :
+        SBInterpolatedImpl(kInterp, stepk, 0., gsparams), //fill in maxk below
+        _dk(dk)
+    {
+        dbg<<"stepk = "<<stepk<<std::endl;
+        dbg<<"kimage bounds = "<<realKImage.getBounds()<<std::endl;
+        assert(_kInterp.get());
+
+        Ninitial = std::max(realKImage.getXMax()-realKImage.getXMin()+1,
+                            realKImage.getYMax()-realKImage.getYMin()+1);
+        dbg<<"Ninitial = "<<Ninitial<<std::endl;
+        Nk = goodFFTSize(int(Ninitial));
+        dbg<<"Nk = "<<Nk<<std::endl;
+
+        _ktab = boost::shared_ptr<KTable>(new KTable(Nk, _dk));
+        _maxk = Ninitial/2 * _dk;
+        dbg<<"_dk = "<<_dk<<std::endl;
+        dbg<<"_maxk = "<<_maxk<<std::endl;
+
+        // Only need to fill in x>=0 since the negative x's are the Hermitian
+        // conjugates of the positive x's.
+        int kxStart = 0;
+        int ikxStart = (realKImage.getXMin()+realKImage.getXMax()+1)/2;
+        int ky = -((realKImage.getYMax()-realKImage.getYMin()+1)/2);
+        dbg<<"kxStart = "<<kxStart<<", kyStart = "<<ky<<std::endl;
+        for (int iky = realKImage.getYMin(); iky<= realKImage.getYMax(); ++iky, ++ky) {
+             int kx = kxStart;
+             for (int ikx = ikxStart; ikx<= realKImage.getXMax(); ++ikx, ++kx) {
+                 std::complex<double> kvalue(realKImage(ikx, iky), imagKImage(ikx, iky));
+                 _ktab->kSet(kx, ky, kvalue);
+                 xxdbg<<"ikx,iky,kx,ky = "<<ikx<<','<<iky<<','<<kx<<','<<ky<<std::endl;
+                 xxdbg<<"kvalue = "<<kvalue<<std::endl;
+             }
+        }
+        _flux = kValue(Position<double>(0.,0.)).real();
+        dbg<<"flux = "<<_flux<<std::endl;
+
+        xcentroid = NAN;
+        ycentroid = NAN;
```

Comment:
Let's not use NAN as a flag that these haven't been set yet.  The support for NAN, and particularly isnan, in C and C++ has kind of a sordid history.  I'm not confident that `std::isnan` will work correctly on all compilers and OS's.  

Better to just set a bool variable `centroidIsDefined = false` here, and check for that later.